### ISSUE TITLE
`hugo server` doesn't stop on Ctrl+C if theme name is misspelled after starting the server

### ISSUE THREAD

**Issue by rootkea** [gohugoio/hugo#8340]
Hello!

`hugo server` says "Press Ctrl+C to stop" but the server doesn't stop if theme name is misspelled after starting the server. The only way to exit the `hugo server` process then is to close the terminal forcefully.

<pre>
$ hugo version
hugo v0.81.0-59D15C97+extended linux/amd64 BuildDate=2021-02-19T17:07:12Z VendorInfo=gohugoio
</pre>

### Steps to reproduce:
1. Start `hugo server` with correct `theme: <theme_name>` in config
2. Now change `<theme_name>` to some non-existent theme name (e.g. misspelling)
3. Now try to stop server with `Ctrl+C`

**Comment by rootkea** [comment]
Doesn't matter even if the spelling is corrected later. Ctrl+C just doesn't stop the `hugo server` process.

**Comment by davidofferman** [comment]
I started looking into this. It started with version v0.80.0 with commit cea157402365f34a69882110a4208999728007a6

**Comment by Vimux** [comment]
I (accidentally) reproduced this bug on my development machine this morning.
Not critical, but very annoying I would say. 😄 

### What version of Hugo are you using (`hugo version`)?

```
$ hugo version
hugo v0.82.0+extended darwin/amd64 BuildDate=unknown
```

### Does this issue reproduce with the latest release?

Yes.

### Steps to reproduce

1. Start `hugo server`
1. Add `!` sign as a line somewhere in config (more correct: add or remove something that would cause ERROR)
1. Try to stop server with Ctrl+C

**Comment by danisztls** [comment]
I had a similar problem. It looks like any error that would in a "fail to reload config" would  break CTRL-C expected behavior.

```shell
Change of config file detected, rebuilding site.
2021-07-07 22:02:24.482 -0300
ERROR 2021/07/07 22:02:24 Failed to reload config:

ERROR 2021/07/07 22:02:24 Module "github.com/twbs/bootstrap" not found; either add it as a Hugo Module or store it in "/home/me/Dev/work/martin/symbols-doc/themes".: module does not exist
Rebuilt in 2 ms
^C^C^C^C^C^C^C^C^C^C^C^C
Change of config file detected, rebuilding site.
2021-07-07 22:02:59.482 -0300
Rebuilt in 886 ms
^C^C^C^C^C^C^C^C^Czsh: killed     hugo server --disableFastRender --environment testing
```

**Comment by MasterGroosha** [comment]
I'm having the same issue as @danisztls (Ctrl+C stops working on ANY error) with **v0.89.4** on Manjaro

**Comment by jmooring** [comment]
Unable to reproduce with v0.105.0. Fixed in v0.99.0 with https://github.com/gohugoio/hugo/pull/9899.
